### ISSUE TITLE
Inspelningens maskering missar common.describe — visningsnamn i klartext

### ISSUE THREAD

**Issue by davidyakoop88-hub** [davidyakoop88-hub/vyra-production#357]
Maskeringen i `tiktok-bridge/inspelare.js` täcker fältnamnen `nickname`, `name`, `username`,
`displayname` — men **inte** `common.describe`, som innehåller samma visningsnamn i klartext.

## Uppmätt

I inspelningen från 2026-09-06 (2046 händelser, 28 minuter) bär `nyttolast.common.describe`
**68 unika strängar** av formen:

```
"<visningsnamn>: gifted the host 1 <gåva>"
"<visningsnamn>: gifted <annat visningsnamn> 10 <gåva>"
```

Alltså både avsändarens och mottagarens riktiga visningsnamn, oförvanskade — i en fil vars
filhuvud säger `"maskad":true`.

`inspelare.js:33`:

```js
const NAMN_FALT = new Set(['nickname', 'name', 'username', 'displayname']);
```

`describe` står inte där. Fältet är en färdigformaterad mening från TikTok, så maskeringen som
letar efter *fältnamn* ser den aldrig.

## Varför det spelar roll

Maskeringen är hela skälet till att en inspelning går att dela vidare — skicka till en utvecklare,
bifoga i en issue, spara som provfixtur. `SPELA-IN-TIKTOK.cmd` säger uttryckligen att filerna är
maskerade och att man ska titta i dem innan man delar dem. Den som läser det, ser
`"maskad":true` i filhuvudet och litar på det delar tittarnas namn utan att veta om det.

Det gäller även **gyllene-payload-korpusen** (#354): planen är att checka in en riktig maskerad
inspelning som fixtur. Med den här luckan skulle namnen hamna i repot.

## Förslag

Maskera `describe` också. Fältet är fritext, så det går inte att ersätta per fältnamn — men det är
härlett: strängen består av visningsnamn plus en mall. Två vägar:

1. **Ersätt hela `describe`** med en platshållare. Enklast, och fältet används inte av någon
   konsument i VYRA (bryggan läser `giftDetails`, inte `describe`).
2. **Ersätt de namn som redan är kända** i samma payload — varje namn som maskeras i ett annat
   fält byts mot samma hash även inne i `describe`. Behåller meningens form för felsökning.

Alternativ 1 är sannolikt rätt: ingenting läser fältet, och en maskering som måste förstå TikToks
meningsmallar blir fel så fort mallen ändras.

⚠️ **Tills detta är åtgärdat:** behandla filer i `tiktok-bridge/inspelningar/` som om de innehåller
personuppgifter, även när filhuvudet säger `maskad: true`. De är gitignorerade, vilket är rätt —
men de får inte bifogas i en issue eller checkas in som fixtur.

## Hur det hittades

Under analysen av 2026-09-06-inspelningen letade jag efter dekorativa Unicode-namn för att mäta
premissen bakom NFKC-fixen (#342/#351). `describe` var det enda fältet som hade namn kvar att mäta
på — vilket i sig var svaret.

**Comment by davidyakoop88-hub** [comment]
**Läckaget är större än issuen beskriver.** Uppmätt över hela inspelningen från 2026-09-06
(27 MB, 3697 rader):

## Maskeringen fungerar — på de fält den känner till

```
nickname maskerade:      5830
nickname i klartext:        0
```

Den delen är alltså inte trasig. Problemet är vad listan **inte** täcker.

## Råa id-fält som går igenom

**2972 unika 15–22-siffriga id:n i klartext**, i fält som inte står i `NAMN_FALT`:

| Fält | Förekomster |
|---|---|
| `currentSponsorId` | 915 |
| `uid` | 587 |
| `toUserId` | 188 |
| `userIdStr` | 154 |
| `anchorId` | 71 |
| `contributorId` | 25 |
| `sendUserId` | 14 |

Maskeringen ersätter `userId` med `id#hash`, men samma person förekommer parallellt som `uid`
eller `userIdStr` i andra delar av payloaden — i klartext. Hashen är alltså inte en mask, den är
ett *alias* bredvid originalet.

## Avatar-URL:er

**13 unika omaskerade CDN-URL:er.** En avatarlänk pekar ut en person lika säkert som ett namn —
den går att öppna.

## Och `describe`, som issuen redan beskriver

142 strängar med visningsnamn i klartext.

## Vad det betyder för åtgärden

Förslaget i issuen — maskera `describe` — räcker inte. Maskeringen behöver byta princip:

- **I dag:** en lista över fältnamn som ska maskeras. Allt som inte står där släpps igenom.
- **Behövs:** en lista över vad som får släppas igenom, och allt annat maskeras eller tas bort.
  Alltså samma vändning som Dockerfilens dokumentrot redan gjort: *"vad som når dokumentroten
  bestäms av en allowlist i stället för att kopiera repot och radera efteråt."* Skälet är
  identiskt — en fil som aldrig kopierades kan inte serveras, och ett fält som aldrig skrevs kan
  inte läcka.

En mellanväg som är billig och stänger det mesta: maskera **varje värde som matchar `\d{15,22}`**
oavsett fältnamn, och varje `https://…tiktokcdn…`-URL. Det tar id:n och avatarer utan att förstå
payloadens form.

## Konsekvens för #354

Gyllene-payload-korpusen planerar att checka in en riktig inspelning som fixtur. **Det går inte
förrän det här är åtgärdat** — och när det är det bör fixturen genereras om från grunden, inte
maskeras i efterhand.

⛔ Tills vidare: inspelningarna i `tiktok-bridge/inspelningar/` innehåller personuppgifter trots
`"maskad":true` i filhuvudet. Gitignore skyddar mot incheckning, inget skyddar mot att filen
skickas vidare.

**Comment by davidyakoop88-hub** [comment]
## Rättelse: min egen eskaleringskommentar hade fel på tre punkter

Jag har nu mätt **alla nio inspelningarna** (58 876 rader, 417 MB) i stället för att räkna på en. Tre av siffrorna och en av rekommendationerna i kommentaren ovan håller inte.

### 1. ⛔ Förslaget "maskera varje `\d{15,22}`" hade förstört inspelaren

Det var min mellanväg. Den är fel, och det är det viktigaste fyndet här.

`score` är en **sträng** i payloaden, inte ett tal — **7 304 unika värden**. `msgId` är också en sträng: **31 132 unika**. En regel på sifferform hade maskerat båda, plus `battleId`, `roomId` och `privilegeId`.

`score` är exakt den data inspelaren finns för. Rekommendationen hade tagit bort battle-mätningen för att skydda personuppgifter som inte ligger där.

**Sifferform kan inte skilja ett person-id från en battle-poäng. Bara fältnamnet kan det.**

### 2. "2 972 unika råa id" — verkligt tal är **416**

Jag räknade `msgId` och `score` som id. De faktiska person-id:n, över alla nio filerna:

| Fält | Förekomster | Unika |
|---|---|---|
| `fromUserId` | 676 | 221 |
| `idStr` | 31 853 | 217 |
| `userIdStr` | 11 629 | 172 |
| `contributorId` / `contributorIdStr` | 253 / 253 | 85 / 85 |
| `currentSponsorId` | 9 602 | 7 |
| `toUserId` | 586 | 4 |
| `uid` | 13 763 | 1 |
| `anchorId` | 586 | 1 |
| `sendUserId` | 33 | 1 |

**416 unika över 10 fält.** Ytterligare 155 i 20 fält behöver bedömas ett i taget (`envelopeId`, `opUid`, `actionByUserId`, `sponsorId`, `linkerId`, `anchorIdStr` …).

### 3. "13 omaskerade CDN-/avatar-URL:er" — **noll** URL:er läcker

Den befintliga regeln `/^https?:\/\//` reducerar varje URL till sitt origin, och det gör den överallt. De 13 jag såg var *origins* (`https://p16-sign-va.tiktokcdn.com/…`), som inte pekar ut någon.

**Men den verkliga avatarläckan är större än den jag rapporterade, och ligger någon annanstans:**

Fältet `uri` bär **3 130 unika värden**, varav **76 332 förekomster är avatarsökvägar** (`.../avt-.../...`). De saknar `https://`, så URL-regeln ser dem aldrig. En avatarsökväg pekar ut en person lika säkert som en full URL — den går att sätta ihop till en.

### Hela läckagelistan, mätt

| Vad | Fält | Unika |
|---|---|---|
| Person-id i klartext | 10 | 416 |
| Avatarsökvägar (`uri`, utan schema) | 1 | ~3 130 |
| `describe` med visningsnamn | 1 | 361 |
| `showArgs` (alla innehåller `:` — TikToks argumentform) | 1 | 1 068 |
| `contributorDisplayId` (display-id = användarnamn) | 1 | 87 |
| `openWebUrl` / `openweburl` (djuplänkar) | 2 | 633 |

### Vad inversionen faktiskt kostar

Jag mätte det innan jag föreslår den. Strängvärden som varken är rena tal, URL:er eller redan maskerade:

- **FRITEXT** (blanksteg eller icke-ASCII — där namn bor): 41 fält, 5 487 unika
- **TOKEN** (kort ren ASCII utan blanksteg — enum, sökväg, flagga): 58 fält, 24 092 unika, varav `logId` ensamt står för 21 609

### Reviderat förslag

Inversionen står fast, men **skiktad efter värdets form, inte efter en lista över farliga fältnamn**:

1. Känt personfält → hasha (listan ovan, utökad)
2. Känt namn-/textfält → maskera (som i dag)
3. **Rena siffror → SLÄPP IGENOM** ← detta skyddar `score`, `msgId`, `battleId`, tidsstämplar
4. `https?://` → origin (som i dag)
5. **Allt annat → maskeras**, med en allowlist av *bevisat ofarliga fältnamn* som växer när ett fält granskats

Steg 5 är vändningen: `describe`, `uri`, `showArgs` och varje framtida fritextfält fastnar utan att någon behöver namnge dem först. Ett okänt fält behåller sin **nyckel och sin form** (`namn#hash`, `<text N tecken>`), så inspelaren kan fortfarande upptäcka att fältet finns — och man befordrar det till allowlisten efter att ha tittat på det. Samma princip som Dockerfilens dokumentrot.

⚠️ Kvarstående risk som inte går att designa bort: `"john_doe"` och `"gift_streak"` går inte att skilja åt på form. Därför måste allowlisten i steg 5 gå på **fältnamn**, aldrig på mönster — ett mönster som släpper igenom korta ASCII-tokens hade släppt igenom användarnamn.

⛔ Spärren mot #354 står kvar, och blir starkare: en fixtur får genereras först när det här är på plats, och då från grunden.

**Comment by davidyakoop88-hub** [comment]
Stängd av \`fec897e\` (PR #376).

Maskeringens förval är nu **maskera**, och verifierat mot de nio riktiga inspelningarna: noll läckage i alla sju kategorierna över 5 954 813 strängvärden, med battle-datan intakt.

Arbetet blev bredare än issuen beskrev, och tre av fynden fanns inte i den:

1. **`uri` bar avatarsökvägar utan `https://`** — 76 332 förekomster som URL-regeln aldrig såg.
2. **24 råa id låg som objektnycklar** i `armies`, `battleComboV2`, `leagueInfoMap` och `userInfos`. Varken issuen eller min första mätning fångade dem — båda letade i värden, medan maskeringen bevarade nycklar med flit.
3. **Min första inversion hade en egen läcka:** 91 värden i `score` och 13 i `key` är samma strängar som står som `userId` någon annanstans. De passerade eftersom jag lät alla rena siffror gå igenom.

Två rättelser till det jag själv skrev i eskaleringen ovan står kvar som viktiga: förslaget att maskera varje `\d{15,22}` hade **förstört inspelaren** (`score` är en sträng på upp till 19 siffror, `msgId` exakt 19), och de "13 omaskerade CDN-URL:erna" var origins, som inte pekar ut någon.

**Kvar för dig, och det är därför #354 fortfarande är spärrad:** de nio befintliga inspelningarna är `maskeringVersion` 1 och läcker trots `maskad: true`. Den här ändringen gör dem inte säkra i efterhand. Rekommendation: radera dem och spela in nytt när du ändå kör med `VYRA_INSPELNING_TYPER=alla` (punkt 2 i `docs/lansering.md`). Fixturen till #354 ska då genereras från grunden, inte maskeras i efterhand.
